### PR TITLE
[Bugfix:Submission] Fixes gradeable relocking

### LINKED ISSUE #9037: Relocking the gradeable doesn't work
**Describe the bug**
Relocking the unlocked gradable doesn't work

**Expected behavior**
Relocking the unlocked gradable is supposed to lock the gradable

**To Reproduce**
1. Create a new gradable
2. Under the submissions/auto-grading, Lock the gradable by choosing one of the dropdowns and set the auto grader points to be a reasonable value  
3. Unlock the gradable
4. Try relocking the gradable
5. See error

**Configuration**
 - OS: Linux
 - Browser safari

**Screenshots**
image

### PR DESCRIPTION
**Pull request by Suryansh5545**
### Please check if the PR fulfills these requirements:

* [x] Tests for the changes have been added/updated (if possible)
* [x] Documentation has been updated/added if relevant

### What is the current behavior?
Fixes #9037 The gradeable was not being re locked. it was though to be only limited to that but since the page change only sended depend_on or depend_on_points at a time the locking function wasn't working. 

### What is the new behavior?
Both the data values are send at the same time and the depend_on_point also get checked for null value in the controller.
Below is a gif of the implementation working

RelockingFixed

### CODE REVIEW

File: site/app/controllers/admin/AdminGradeableController.php [hunk 1]
```diff
@@ -1133,7 +1133,7 @@ private function updateGradeable(Gradeable $gradeable, $details) {
                             $gradeable->setDependsOnPoints(0);
                         }
                         else {
-                            if ($depends_on_points < 0 || $depends_on_points > $temp_gradeable->getDependsOnPoints()) {
+                            if ($depends_on_points < 0 || ($depends_on_points > $temp_gradeable->getDependsOnPoints() && $temp_gradeable->getDependsOnPoints() != null)) {
```
Comment (williamjallen) [inline site/app/controllers/admin/AdminGradeableController.php:1136]: It is better to use the strict comparison operators in PHP.

```suggestion
                            if ($depends_on_points < 0 || ($depends_on_points > $temp_gradeable->getDependsOnPoints() && $temp_gradeable->getDependsOnPoints() !== null)) {
```

File: site/public/js/admin-gradeable-updates.js [hunk 2]
```diff
@@ -223,6 +223,12 @@ $(document).ready(function () {
             }
             data[val.name] = $(val).val();
         };
+        if (data['depends_on'] != null) {
+            data['depends_on_points'] = points;
+        }
+        else if (data['depends_on_points'] != null) {
+            data['depends_on'] = gradeable;
+        }
```
Comment (williamjallen) [inline site/public/js/admin-gradeable-updates.js:231]: same here.
```suggestion
        if (data['depends_on'] !== null) {
            data['depends_on_points'] == points;
        }
        else if (data['depends_on_points'] != null) {
            data['depends_on'] = gradeable;
        }
```

**Review by williamjallen — CHANGES_REQUESTED** [review]

**Review by ch3nnyj — APPROVED** [review]
I've tested your branch and verified that the issue no longer appears when re-locking a gradeable. I see no issues in your code as well. Maybe William might find something but overall it looks like you fixed the issue.

File: site/public/js/admin-gradeable-updates.js [hunk 3]
```diff
@@ -223,6 +223,12 @@ $(document).ready(function () {
             }
             data[val.name] = $(val).val();
         };
+        if (data['depends_on'] !== null) {
+            data['depends_on_points'] == points;
```
Comment (williamjallen) [inline site/public/js/admin-gradeable-updates.js:227]: Is the `==` operator here a typo?

**Review by williamjallen — CHANGES_REQUESTED** [review]

File: site/public/js/admin-gradeable-updates.js (hunk 3 above)
Comment (Suryansh5545) [reply site/public/js/admin-gradeable-updates.js:227]: It was in the suggested change in last conversation and the functionality still worked so I changed both. I will revert it.

Comment (Suryansh5545) [reply site/public/js/admin-gradeable-updates.js]: @williamjallen it was in the last suggested change and didn't break the functionality. So I changed both from != and = to !== and == . I have changed it back now.

**Review by ch3nnyj — APPROVED** [review]
Functionality didn't change so I must have missed that type on my end when reviewing. I should be all set now to merge.

**Pull request merged**